ttnn/ttnn/operations/core.py

```diff
@@ -779,9 +788,19 @@ def impl(input_tensor):
         *batch_sizes, _, _ = tensor.shape
 
         if is_on_device:
-            tensor = ttl.tensor.tilize_with_val_padding(
-                tensor, batch_sizes + [padded_height, padded_width], [0, 0, 0, 0], 0, output_dtype=dtype
-            )
+            if memory_config is None:
```

Comment:
Same here. Why do you need if-else statement? Pass in the config directly in either case